### PR TITLE
UI handles preflights skipped with unattended installs

### PR DESCRIPTION
**Pull request by divolgin**
Screen Shot 2021-06-02 at 8 54 00 PM

### CODE REVIEW

File: web/src/components/snapshots/SnapshotStorageDestination.jsx [hunk 1]
```diff
@@ -354,7 +355,7 @@ class SnapshotStorageDestination extends Component {
         await this.snapshotProviderGoogle();
         break;
       case "other":
-        const s3CompatibleFieldErrors = this.validateSnapshotProviderS3Compatible();
```
Comment (divolgin) [inline web/src/components/snapshots/SnapshotStorageDestination.jsx:357]: This change is unrelated.  JSX compiler was complaining about it.

**Review by GraysonNull — APPROVED** [review]

File: pkg/store/kotsstore/migrations.go [hunk 2]
```diff
@@ -36,6 +36,10 @@ func (s *KOTSStore) RunMigrations() {
 		logger.Error(errors.Wrap(err, "failed to migrate app_spec"))
 	}
 
+	if err := s.migrationSkippedPreflights(); err != nil {
```
Comment (salahalsaleh) [inline pkg/store/kotsstore/migrations.go:39]:
```suggestion
	if err := s.migrateSkippedPreflights(); err != nil {
```

File: web/src/components/PreflightResultPage.jsx [hunk 3]
```diff
@@ -312,9 +312,10 @@ class PreflightResultPage extends Component {
     const { slug } = this.props.match.params;
     const { showSkipModal, showWarningModal, preflightResultData, errorMessage } = this.state;
 
-    const stopPolling = !!preflightResultData?.result;
+    const preflightSkipped = preflightResultData?.skipped;
+    const stopPolling = (preflightResultData?.result || preflightSkipped);
     let preflightJSON = {};
-    if (stopPolling) {
+    if (stopPolling && !preflightSkipped) {
```
Comment (salahalsaleh) [inline web/src/components/PreflightResultPage.jsx:318]: how about doing this instead?
```suggestion
    if (preflightResultData?.result) {
```

**Review by salahalsaleh — APPROVED** [review]

File: web/src/components/PreflightResultPage.jsx (hunk 3 above)
Comment (divolgin) [reply web/src/components/PreflightResultPage.jsx:318]: hmm, you don't think this is easier to read?
